scripts/periodic/check_crossref_dois.py

```diff
@@ -0,0 +1,78 @@
+import sys
+import logging
+import requests
+
+from datetime import timedelta
+
+from framework.celery_tasks import app as celery_app
+from website import settings
+from website import mails
+from website.app import init_app
+from django.utils import timezone
+
+import django
+django.setup()
+
+from osf.models import Preprint
+
+from scripts.utils import add_file_logger
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def check_crossref_dois(dry_run=True):
+    """
+    This script is to check for any DOI confirmation messages we may have missed during downtime and alert admins to any
+    DOIs that have been pending for X number of days.
+    :param dry_run:
+    :return:
+    """
+    # Create one enormous url to check if all our pending crossref DOIs are good, then set all identifers
+    preprints_with_pending_dois = Preprint.objects.filter(preprint_doi_created__isnull=True,
+                                                      is_published=True)
+
+    pending_dois = []
+
+    for preprint in preprints_with_pending_dois:
+        prefix = preprint.provider.doi_prefix
+        pending_dois.append('doi:{}'.format(settings.DOI_FORMAT.format(prefix=prefix, guid=preprint._id)))
+
+    url = '{}/works?filter={}'.format(settings.CROSSREF_JSON_API_URL, ','.join(pending_dois))
+    preprints = requests.get(url).json()['message']['items']
```

Comment:
What happens when this fails? Does the script just not run for the day? If so, maybe include a retry mechanism (and test for this condition).

Reply:
A retry mechanism is built in to requests, but I will improve the exception handling here to log and throw the error better.